portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,26 +32,45 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> messages = new ArrayList<String>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String messagesJson = convertToJson(messages);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
same here

Reply:
Here I record timestamp so I can sort the comments by time (first line of doGet())

Reply:
This is the "read"/GET function though. I don't think removing this line would change the behavior of this function at all. The variable `timestamp` is not used in this function after. Am I missing something?

Reply:
Wait, I commented on the wrong conversation, You're right, this line is unnecessary